libzmix/bulletproofs_amcl/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "bulletproofs_amcl"
+version = "0.1.0"
+authors = ["lovesh harchandani <lovesh.bond@gmail.com>"]
+edition = "2018"
+
+[dependencies]
+rand = "0.6"
+lazy_static = "1.3.0"
+merlin = "1"
+byteorder = "1.3.1"
+
+[dependencies.amcl_wrapper]
```

Comment:
Why can't we use libursa/pair/amcl.rs? Let's update it so we don't have to wrap amcl twice.

Reply:
Few reasons. Code in amcl.rs is annoying to write/read when you have large number of operations due to lack of operator overloading. Secondly, [amcl_wrapper](https://github.com/lovesh/amcl_rust_wrapper) has abstractions for vectors ops which again makes the code easier to write/read. Apart from the abstractions, amcl_wrapper contains optimizations like Barrett reduction, constant time as well as variable time multi-scalar multiplication and batch inversion. These things do not matter for the stuff we do in libursa now so amcl.rs is sufficient there. But these are necessary for the Bulletproof. So consider the wrapper an external dependency.
Also code in amcl.rs is using a much older version of the amcl and there has been numerous fixes since then. We should get a basic Travis CI setup on miracl/amcl that deploys to crates.io. In the meantime, the wrapper uses crate deployed from this [mirror](https://github.com/lovesh/amcl-mirror).